include/albatross/src/models/ransac.hpp

```diff
@@ -289,38 +395,37 @@ class Ransac : public ModelBase<Ransac<ModelType, StrategyType>> {
     auto indexer = strategy_.get_indexer(dataset);
     auto ransac_functions = strategy_(sub_model_, dataset);
 
-    std::vector<FoldName> inliers =
-        ransac(ransac_functions, map_keys(indexer), inlier_threshold_,
-               random_sample_size_, min_consensus_size_, max_iterations_);
-
-    const auto good_inds = indices_from_names(indexer, inliers);
-    const auto consensus_set = subset(dataset, good_inds);
-
-    std::vector<FoldName> outliers;
-    for (const auto &pair : indexer) {
-      if (!contains_group(inliers, pair.first)) {
-        outliers.emplace_back(pair.first);
-      }
+    const auto ransac_output =
+        ransac(ransac_functions, map_keys(indexer), config_.inlier_threshold,
+               config_.random_sample_size, config_.min_consensus_size,
+               config_.max_iterations);
+
+    if (ransac_success(ransac_output.return_code)) {
+      const auto good_inds = indices_from_names(indexer, ransac_output.inliers);
+      const auto consensus_set = subset(dataset, good_inds);
+      return Fit<RansacFit<ModelType, StrategyType, FeatureType>>(
+          sub_model_.fit(consensus_set), ransac_output);
+    } else {
+      return Fit<RansacFit<ModelType, StrategyType, FeatureType>>(
+          ransac_output);
     }
-
-    // Then generate a fit.
-    return Fit<RansacFit<ModelType, StrategyType, FeatureType>>(
-        sub_model_.fit(consensus_set), inliers, outliers);
   }
 
   template <typename PredictFeatureType, typename FitType, typename PredictType>
   PredictType _predict_impl(const std::vector<PredictFeatureType> &features,
                             const FitType &ransac_fit_,
                             PredictTypeIdentity<PredictType> &&) const {
-    return ransac_fit_.fit_model.predict(features).template get<PredictType>();
+    // If RANSAC failed it's up to the user to determine that from the output of
+    // fit and deal with it appropriately.
```

Comment:
This comment says "If RANSAC failed" ... but it seems like this still includes "rc_Success" but with an empty_fit(). Is empty_fit a success or a failure? I'd say failure, since it can't be used to make predications.

Reply:
assuming this is the same issue you pointed out on line 166?  If it's something else let me know.

Reply:
This is the same issue I mentioned before. Thankyou for addressing it :)